test/cpp/api/autograd.cpp

```diff
@@ -154,6 +155,18 @@ TEST(AutogradAPITests, RetainGrad) {
   ASSERT_VARIABLE_EQ(input * 18, input.grad());
 }
 
+TEST(AutogradAPITests, AnomalyMode) {
+  // Needs to have backtrace as warning and then throw an error
+  WarningCapture warnings;
+  torch::autograd::AnomalyMode::set_enabled(true);
+  auto x = torch::tensor({5.0}, torch::requires_grad());
+  auto y = x * x;
+  auto z = y * y;
+  y += 1;
+  ASSERT_THROWS_WITH(z.backward(), "inplace");
+  ASSERT_TRUE(warnings.str().find("Traceback of forward") != std::string::npos);
```

Comment:
Maybe we should add a test for the double backward case just to make sure it works and we get both stack traces?
Something like `test_nested_anomaly_detect_nan` which is in `test_autograd.py` ?

Reply:
Would it be OK to use 0**1.5 and double backward that or do you prefer a custom Function? (The split between AutogradAPITest and CustomAutogradTest makes me wonder.)

Reply:
I added that. We get to backtraces. :slightly_smiling_face: